### PR TITLE
Hotfix/media-search-sort-error

### PR DESCRIPTION
**Pull request by danialeresbar**
Maybe the sort is not necessary...

### CODE REVIEW

File: djangoplicity/archives/contrib/search/forms.py [hunk 1]
```diff
@@ -120,7 +120,6 @@ class AdvancedSearchForm (forms.Form):
 
         # take fields from options class, and sort them correctly
         fields = [(field_name, obj) for field_name, obj in options.AdvancedSearch.__dict__.items() if isinstance(obj, formsFields.Field)]
-        fields.sort(lambda x, y: cmp(x[1].creation_counter, y[1].creation_counter))
```
Comment (danialeresbar) [inline djangoplicity/archives/contrib/search/forms.py:124]: Th "sort" line should be here

`fields.sort(key=lambda field: field[1].creation_counter)`

But I get an Attribute error even though the class should have the attribute (Inheritance) :(

image

I have tried many ways but I can't figure out why 😣

Comment (danialeresbar) [inline djangoplicity/archives/contrib/search/forms.py:124]: Now. If we remove the "sort" the error is not displayed:

image

I did some searches (images) on my local and on the space site in order to find differences. However, there were no differences in searches:

image

image

image

image

What do you think?

Comment (EdisonArango) [reply djangoplicity/archives/contrib/search/forms.py:124]: What's creation_counter?. Did you check the ImageSizeSearchField?

Comment (danialeresbar) [reply djangoplicity/archives/contrib/search/forms.py:124]: Yes, I did it. I checked the ImageSizeSearchField class in order to review the inheritance members:

ImageSizeSearchField inherits from SearchField, SearchField inherits from Field (django). Field has the "creation_counter" attribute, but I don't understand why ImageSearchField doesn't have this attribute? :(

I checked the builder functions and definitions and each is correct

I will check again today!

File: djangoplicity/archives/contrib/search/fields.py [hunk 2]
```diff
@@ -59,6 +59,7 @@ class SearchField( fields.Field ):
     operator = 'icontains'
     help_text = ''
     logic_repr = {}
+    creation_counter = 0
```
Comment (danialeresbar) [inline djangoplicity/archives/contrib/search/fields.py:62]: Added attribute

File: djangoplicity/archives/contrib/search/fields.py [hunk 3]
```diff
@@ -73,6 +74,10 @@ def __init__( self, model_field='', help_text=None, *args, **kwargs ):
             help_text = self.help_text
         kwargs['help_text'] = help_text
 
+        # Increase the creation counter, and save our local copy.
+        self.creation_counter = SearchField.creation_counter
+        SearchField.creation_counter += 1
+
```
Comment (danialeresbar) [inline djangoplicity/archives/contrib/search/fields.py:80]: Copied attribute behavior from fields.Field

**Comment by EdisonArango** [comment]
Great, thank you!

**Pull request merged**